rxjava-api-client/src/main/java/com/desk/java/apiclient/RxDeskClient.java

```diff
@@ -0,0 +1,236 @@
+import com.desk.java.apiclient.service.FilterService;
+import com.desk.java.apiclient.service.GroupService;
+import com.desk.java.apiclient.service.InboundMailboxService;
+import com.desk.java.apiclient.service.LabelService;
+import com.desk.java.apiclient.service.MacroService;
+import com.desk.java.apiclient.service.OutboundMailboxService;
+import com.desk.java.apiclient.service.PermissionService;
+import com.desk.java.apiclient.service.RxArticleService;
+import com.desk.java.apiclient.service.RxCaseService;
+import com.desk.java.apiclient.service.RxCompanyService;
+import com.desk.java.apiclient.service.RxCustomFieldsService;
+import com.desk.java.apiclient.service.RxCustomerService;
+import com.desk.java.apiclient.service.RxFilterService;
+import com.desk.java.apiclient.service.RxGroupService;
+import com.desk.java.apiclient.service.RxInboundMailboxService;
+import com.desk.java.apiclient.service.RxLabelService;
+import com.desk.java.apiclient.service.RxMacroService;
+import com.desk.java.apiclient.service.RxOutboundMailboxService;
+import com.desk.java.apiclient.service.RxPermissionService;
+import com.desk.java.apiclient.service.RxSiteService;
+import com.desk.java.apiclient.service.RxTopicService;
+import com.desk.java.apiclient.service.RxTwitterUserService;
+import com.desk.java.apiclient.service.RxUserService;
+import com.desk.java.apiclient.service.SiteService;
+import com.desk.java.apiclient.service.TopicService;
+import com.desk.java.apiclient.service.TwitterUserService;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+
+import retrofit.CallAdapter.Factory;
+import retrofit.RxJavaCallAdapterFactory;
+
+/**
+ * <p>
+ *     Client which interfaces with the Desk API and exposes RxJava services.
+ * </p>
+ *
+ * Created by Jerrell Mardis
+ * Copyright (c) 2015 Desk.com. All rights reserved.
+ */
+public class RxDeskClient extends DefaultDeskClient {
+
+    private RxUserService rxUserService;
+    private RxSiteService rxSiteService;
+    private RxLabelService rxLabelService;
+    private RxCustomFieldsService rxCustomFieldsService;
+    private RxGroupService rxGroupService;
+    private RxMacroService rxMacroService;
+    private RxOutboundMailboxService rxOutboundMailboxService;
+    private RxFilterService rxFilterService;
+    private RxCaseService rxCaseService;
+    private RxCompanyService rxCompanyService;
+    private RxCustomerService rxCustomerService;
+    private RxPermissionService rxPermissionService;
+    private RxTwitterUserService rxTwitterUserService;
+    private RxTopicService rxTopicService;
+    private RxArticleService rxArticleService;
+    private RxInboundMailboxService rxInboundMailboxService;
+
+    /**
+     * Creates a {@link DeskClient} using the builder provided.
+     * @param builder the builder to use to build the {@link DeskClient}
+     * @return a {@link DeskClient}
+     */
+    public static DeskClient create(DefaultDeskClient.Builder builder) {
```

Comment:
The return type should be a `RxDeskClient` here or you have to cast the result to an `RxDeskClient`.